### PR TITLE
SQL variables: add a new variable `tidb_enable_parallel_apply`

### PR DESCRIPTION
**Pull request by Joyinqin**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/4819
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by qw4990 — APPROVED** [review]

File: system-variables.md [hunk 1]
```diff
@@ -1150,3 +1150,9 @@ This variable is an alias for _transaction_isolation_.
 - Scope: SESSION | GLOBAL
 - Default value: ON
 - This variable controls whether to use the high precision mode when computing the window functions.
+
+### `tidb_enable_parallel_apply` <span class="version-mark">New in v5.0 GA</span>
```
Comment (TomShawn) [inline system-variables.md:1154]: Please put this variable in the alphabetical order of entire variable list.
```suggestion
### tidb_enable_parallel_apply <span class="version-mark">New in v5.0 GA</span>
```

File: system-variables.md [hunk 2]
```diff
@@ -1150,3 +1150,9 @@ This variable is an alias for _transaction_isolation_.
...
+
+- Scope: SESSION | GLOBAL
+- Default value: 0
+- This variable controls whether to enable the Apply operator concurrency, and the number of concurrency is controlled by the `tidb_executor_concurrency` variable. The Apply operator is used to process correlated subqueries and has no concurrency by default, so the execution speed is slow. Enable the Apply concurrency switch can increase concurrency and speed up execution. Currently, this variable defaults to be disabled.
```
Comment (TomShawn) [inline system-variables.md]:
```suggestion
- This variable controls whether to enable concurrency for the `Apply` operator. The number of concurrencies is controlled by the `tidb_executor_concurrency` variable. The `Apply` operator is used to process correlated subqueries and has no concurrency by default, so the execution speed is slow. Setting this variable value to `1` can increase concurrency and speed up execution. Currently, concurrency for `Apply` is disabled by default.
```

Comment (TomShawn) [reply system-variables.md]: @Joyinqin Please see this comment.

File: system-variables.md [hunk 3]
```diff
@@ -1150,3 +1156,9 @@ This variable is an alias for _transaction_isolation_.
 - Scope: SESSION | GLOBAL
 - Default value: ON
 - This variable controls whether to use the high precision mode when computing the window functions.
+
+### tidb_enable_parallel_apply <span class="version-mark">New in v5.0 GA</span>
+
+- Scope: SESSION | GLOBAL
+- Default value: 0
+- This variable controls whether to enable the Apply operator concurrency, and the number of concurrency is controlled by the `tidb_executor_concurrency` variable. The Apply operator is used to process correlated subqueries and has no concurrency by default, so the execution speed is slow. Enable the Apply concurrency switch can increase concurrency and speed up execution. Currently, this variable defaults to be disabled.
```
Comment (TomShawn) [inline system-variables.md:1164]:
```suggestion
```

**Pull request merged**